runtime/array.c

```diff
@@ -249,6 +250,15 @@ CAMLprim value caml_make_vect(value len, value init)
       for (i = 0; i < size; i++) Field(res, i) = init;
     }
   }
+#ifndef FLAT_FLOAT_ARRAY
+  if (caml_boxed_float_array_warning_active()
+      && Is_block(init)
+      && Is_in_value_area(init)
+      && Tag_val(init) == Double_tag) {
+    fprintf(stderr, "Warning : creating a boxed float array\n");
+    caml_print_current_callstack(10);
```

Comment:
`10` here seems pretty arbitrary. One possibility would be to allow customizing the length of the back trace through the API.